lunes.go

```diff
@@ -0,0 +1,425 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package lunes
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+	"unicode"
+)
+
+var longDayNamesStd = []string{
+	"Sunday",
+	"Monday",
+	"Tuesday",
+	"Wednesday",
+	"Thursday",
+	"Friday",
+	"Saturday",
+}
+
+var shortDayNamesStd = []string{
+	"Sun",
+	"Mon",
+	"Tue",
+	"Wed",
+	"Thu",
+	"Fri",
+	"Sat",
+}
+
+var shortMonthNamesStd = []string{
+	"Jan",
+	"Feb",
+	"Mar",
+	"Apr",
+	"May",
+	"Jun",
+	"Jul",
+	"Aug",
+	"Sep",
+	"Oct",
+	"Nov",
+	"Dec",
+}
+
+var longMonthNamesStd = []string{
+	"January",
+	"February",
+	"March",
+	"April",
+	"May",
+	"June",
+	"July",
+	"August",
+	"September",
+	"October",
+	"November",
+	"December",
+}
+
+var dayPeriodsStd = []string{
+	"AM",
+	"PM",
+}
+
+// Parse translates the localized time value using the [Translate] method, and parses the
+// output to [time.Time] by using the [time.Parse] method.
+//
+// If the given locale does not support any layout element specified on the layout argument,
+// it results in an ErrUnsupportedLayoutElem error. On the other hand, if the values does
+// not match the layout, an ErrLayoutMismatch is returned.
+// In addition to that, it might return any [time.ParseInLocation] errors.
+func Parse(layout string, value string, locale Locale) (time.Time, error) {
```

Comment:
can we also provide function which allows you to specify locale in a string format e.g `en-EN`

Reply:
Sure thing, it makes sense. in this case, would you recommend making the Parse & ParseInLocation functions generic:

```go
type ParseableLocale interface {
	*Locale | string
}

func Parse[K ParseableLocale](layout string, value string, locale K) (time.Time, error)
```

or having specific specific functions, such as `func ParseWithDefault(...)` (I'm really bad naming)

Reply:
i would probably avoid generics as it's only 2 ways of providing locale, it's either string or Locale
maybe `Parse` and `ParseWithLocale` but it's also confusing. naming is hard

Reply:
I think it wouldn't be so confusing if we change the `ParseX` functions to use a string locale argument, and add new `ParseWithLocale` and `ParseInLocationWithLocale` function for `lunes.Locale` usages.  they're long names, but IMO are fine.

Reply:
sounds good

Reply:
Done through https://github.com/elastic/lunes/pull/4/commits/80a15f8587e6b299fdca9be7c9b00fa949030073, in summary, it changed:
- `Parse`, `ParseInLocation` and `Translate` functions to use string language tags instead of `Locale`.
- Added `ParseWithLocale`, `ParseInLocationWithLocale`, `TranslateWithLocale` function accepting a `Locale` argument.
- Dropped the `language.Tag` usages on the production code, as the `language.Parse/Make` performance is surprisingly bad. Strings are now first citizens for representing language tags.
- The generator now adds a constant for each language tag on the `tables.go`, e.g `lunes.LocaleEs`, `lunes.LocaleEnUS`.
- Docs updates to reflect the changes

Thank you again for reviewing this PR :)